contracts/base/ModuleManager.sol

```diff
@@ -105,8 +107,22 @@ abstract contract ModuleManager is Storage, Receiver, EIP712, IModuleManagerEven
                 }
                 returndatacopy(0, 0, returndatasize())
                 return(0, returndatasize())
+                }
+            }
+        }
+        /// @solidity memory-safe-assembly
+        assembly {
+            let s := shr(224, calldataload(0))
+            // 0x150b7a02: `onERC721Received(address,address,uint256,bytes)`.
+            // 0xf23a6e61: `onERC1155Received(address,address,uint256,uint256,bytes)`.
+            // 0xbc197c81: `onERC1155BatchReceived(address,address,uint256[],uint256[],bytes)`.
+            if or(eq(s, 0x150b7a02), or(eq(s, 0xf23a6e61), eq(s, 0xbc197c81))) {
+                mstore(0x20, s) // Store `msg.sig`.
+                return(0x3c, 0x20) // Return `msg.sig`.
             }
         }
+        // review
+        // could revert with unsupported func selector / calltype etc
```

Comment:
reverting here with `MissingFallbackHandler(msg.sig)` makes sense to me, it would mimic a contract not having a fallback function in these cases

Reply:
if this error is indeed added, add it after the nft callbacks assembly block, otherwise it would also revert for these token callbacks if they don't have a specific handler.

Reply:
yes that's why I removed receiver base. so this will come at very end.

Reply:
please check @chinmay-farkya

Reply:
All assembly blocks return the execution, and if none of them is entered, then it reverts at the end. Looks good. @livingrockrises

Reply:
I have also moved code from this PR https://github.com/bcnmy/nexus/pull/131/files
here on this PR.
@chinmay-farkya